### PR TITLE
Fix memory leaks

### PR DESCRIPTION
**Pull request by ahoppen**

### CODE REVIEW

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test macOS

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

File: Tests/SKCoreTests/BuildSystemManagerTests.swift [hunk 1]
```diff
@@ -429,7 +429,7 @@ class ManualBuildSystem: BuildSystem {
 
   var map: [DocumentURI: FileBuildSettings] = [:]
 
-  var delegate: BuildSystemDelegate? = nil
+  weak var delegate: BuildSystemDelegate? = nil
```
Comment (bnbarham) [inline Tests/SKCoreTests/BuildSystemManagerTests.swift:432]: `Tests/SourceKitLSPTests/BuildSystemTests.swift` has one too 😅. Can't help but think we're going to continue having this issue.

**Review by bnbarham — APPROVED** [review]

File: Tests/SKCoreTests/BuildSystemManagerTests.swift (hunk 1 above)
Comment (ahoppen) [reply Tests/SKCoreTests/BuildSystemManagerTests.swift:432]: But that’s already `weak`. Or am I missing something?

https://github.com/apple/sourcekit-lsp/blob/1fe79781dfd7ca7024b0f6ccd87ef759fe133870/Tests/SourceKitLSPTests/BuildSystemTests.swift#L30

Comment (bnbarham) [reply Tests/SKCoreTests/BuildSystemManagerTests.swift:432]: Only that it was late and I assume I read `weak` as `public` or something?

**Pull request merged**